### PR TITLE
Refactor setup flow and add husky and lint-staged

### PR DESCRIPTION
**Pull request by kaisermann**
#### What is the purpose of this pull request?

- Add `husky` as a dependency and property in the `package.json` (configures git hooks)
- Add `lint-staged` as a dependency and property in the `package.json` (configures linting tasks when committing)
- Refactor a bit the setup flow: 
  - Silence the output (but not errors) of `yarn add ...`, no need for that terminal pollution
  - Moved constants to `consts.js`
  - Renamed `setupESLint.ts` to `setupTooling.ts`

#### What problem is this solving?

The `husky` and `lint-staged` packages are part of our movement of automating and enforcing a standard coding style in VTEX. The rest is an attempt to make the code more readable and decoupled, therefore more maintainable.

#### How should this be manually tested?

1) Create a `alias` in your shell to `$TOOLBELT_PATH/lib/cli.js`

```fish
# fish example
alias vtest=~/Projects/vtex/toolbelt/lib/cli.js
```

2) Execute the `yarn watch` script
3) Go to an app directory and execute the new alias

```bash
$ vtest setup
```

#### Screenshots or example usage

**Current log:**
image

**Proposed log:**
image

I focused on refactoring only the `setupESLint`/`setupTooling` part, the log of both `setupTSConfig` and `setupTypings` are unchanged.

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: src/modules/setup/setupTypings.ts [hunk 1]
```diff
@@ -97,13 +99,13 @@ const injectTypingsInPackageJson = async (appDeps: Record<string, any>, ignoreLi
...
 
-export const setupTypings = async (manifest: Manifest, ignoreLinked: boolean, buildersToAddTypes: string[]) => {
+export const setupTypings = async (manifest: Manifest, ignoreLinked: boolean) => {
   const appName = manifest.vendor + '.' + manifest.name
   const appMajor = toMajorRange(manifest.version)
 
   const { builder: builderClient } = createClients({}, { retries: 2, timeout: 2000 })
   const builders = R.keys(R.prop('builders', manifest) || {})
-  const filteredBuilders = R.intersection(builders, buildersToAddTypes)
+  const filteredBuilders = R.intersection(builders, BUILDERS_WITH_TYPES)
```
Comment (tiagonapoli) [inline src/modules/setup/setupTypings.ts]: Maybe it can be a constant but I think this could make more difficult to test this function

Comment (kaisermann) [reply src/modules/setup/setupTypings.ts]: 🤔 🤔🤔  thinking  🤔 🤔 🤔

Comment (kaisermann) [reply src/modules/setup/setupTypings.ts]: agreed!

**Comment by tiagonapoli** [comment]
Just missing changelog

Awesome work :rocket:

**Pull request merged**